### ISSUE TITLE
check_if_ccache_file_is_used() might fail if ccache is on NFS

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#686]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/682
- Created at 2010-11-16 12:38:50 by [sbose](https://pagure.io/user/sbose)
- Closed as Invalid
- Assigned to nobody
---

check_if_ccache_file_is_used() is called from the krb5 provider with root privileges and will fail is the ccache file is on NFS where root access is not allowed. The same is true for check_for_valid_tgt(). To solve this the ccache file needs to be checked by the krb5_child with the privileges of the corresponding user.

The solution can be implemented in two different ways:

- instead of calling check_if_ccache_file_is_used() and check_for_valid_tgt() a krb5_child is forked and returns the relevant data. This will not change the workflow in the krb5 provider during authentication/change password/access control operations, but will fork krb5_child two times during authentication.

- all the checks are move from the krb5 provider to krb5_child. With this solution krb5_chikd is forked only once for an authentication, but the workflow is changed considerable and a lot of tests are needed to avoid regressions.

### Comments

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2010-11-19 13:07:43](https://pagure.io/SSSD/sssd/issue/682#221061)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.6.0
priority: major => minor

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2011-02-07 15:18:08](https://pagure.io/SSSD/sssd/issue/682#221062)**

Fields changed

coverity:  => 
milestone: SSSD 1.6.0 => SSSD 1.7.0
upgrade:  => 0

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2011-07-22 21:58:36](https://pagure.io/SSSD/sssd/issue/682#221063)**

Fields changed

milestone: SSSD 1.8.0 => SSSD 1.9.0
patch:  => 0

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2012-01-16 16:44:05](https://pagure.io/SSSD/sssd/issue/682#221064)**

Fields changed

blockedby:  => 
blocking:  => 
milestone: SSSD 1.9.0 => SSSD Kerberos improvements
rhbz:  =>

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2012-02-10 23:46:17](https://pagure.io/SSSD/sssd/issue/682#221065)**

Fields changed

rhbz:  => 0

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2012-08-16 23:41:18](https://pagure.io/SSSD/sssd/issue/682#221066)**

Having ccache on the NFS is not a good practice. I do not think it makes sense to ever do. With the recent changes in Fedora to consolidate ccache in .../run/user makes this even less relevant. The direction we expect is to have secure NFS using Kerberos to be deployed more and more. In this case ccache would not be on the NFS anyways.
Moving to NEEDS_TRIAGE with the intent to close or defer. 

feature_milestone:  => 
proposed_priority:  => Undefined

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2012-08-16 23:41:46](https://pagure.io/SSSD/sssd/issue/682#221067)**

Fields changed

milestone: SSSD Kerberos Improvements Feature => NEEDS_TRIAGE

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2012-08-23 16:01:28](https://pagure.io/SSSD/sssd/issue/682#221068)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD Kerberos Improvements Feature
resolution:  => wontfix
status: new => closed

---

**Comment from [sbose](https://pagure.io/user/sbose) at [2017-02-24 14:23:06](https://pagure.io/SSSD/sssd/issue/682#221069)**

**Metadata Update from @sbose**:
- Issue set to the milestone: SSSD Kerberos Improvements Feature
